### PR TITLE
Implements #713 Ublas to Petsc interface

### PR DESCRIPTION
**Pull request by prudhomm**
implement creating a VectorPetsc out of a VectorUblas without copying
@vincentchabannes @prj-

### CODE REVIEW

File: feel/feelalg/vectorpetsc.hpp [hunk 1]
```diff
@@ -170,6 +171,23 @@ class VectorPetsc : public Vector<T>
...
+    template<typename Storage>
+    VectorPetsc( VectorUblas<T,Storage>& v )
+        :
+        super( v.mapPtr() ),
+        M_destroy_vec_on_exit( false )
+        {
+            VecCreateMPIWithArray( v.map().worldComm().globalComm(),
+                                   1, v.map().nLocalDofWithoutGhost(), v.map().nDof(),
+                                   std::addressof( v[0] ), &M_vec );
+            this->M_is_initialized = true;
```
Comment (vincentchabannes) [inline feel/feelalg/vectorpetsc.hpp]: @prudhomm I think it's not good in parallel. the size of M_vec is nLocalDofWithoutGhost() but v is nLocalDofWithGhost(). And the ordering is not the same.

Comment (prudhomm) [inline feel/feelalg/vectorpetsc.hpp]: what do you mean by ordering ?

Comment (prudhomm) [inline feel/feelalg/vectorpetsc.hpp]: actually there is another function that would include the ghosts
http://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Vec/VecCreateGhostWithArray.html#VecCreateGhostWithArray

Comment (prudhomm) [inline feel/feelalg/vectorpetsc.hpp]: so `[0,nLocalDofWithoutGhost()] ` does not correspond to the indices for the local array without the ghosts ?

Comment (vincentchabannes) [inline feel/feelalg/vectorpetsc.hpp]: yes, it is what I mean.
There is an additional Vec M_vecLocal with the same size and ordering. It's used for the container access with local dof index (and allow copy from VectorPetsc to VectorUblas). This vector is updated from M_vec in Localize() method with MpiScatter. This strategy is old (at the begin of feel++ //) and can be rethought.

Comment (vincentchabannes) [inline feel/feelalg/vectorpetsc.hpp]: Maybe we can remove  M_vecLocal and build M_vec with VecCreateGhostWithArray

Comment (prudhomm) [inline feel/feelalg/vectorpetsc.hpp]: but then how about VecCreateGhostWithArray ?

Comment (prudhomm) [inline feel/feelalg/vectorpetsc.hpp]: and actually you can access the ghost part of a `vec` using  `VecGhostGetLocalForm() `

Comment (vincentchabannes) [inline feel/feelalg/vectorpetsc.hpp]: I looked quickly the doc and example of VecCreateGhostWithArray. Indeed, it seems to be the good way (and better that is done actually)

**Comment by prudhomm** [comment]
@vincentchabannes could you have a look at the implementation ? there is currently a crash
do we have access to ghosts global ids ? do we actually need them ?
from petsc's doc we can provide the global indices of ghost points (or NULL if not needed), these do not need to be in increasing order (sorted)

**Comment by prudhomm** [comment]
I am currently testing on the laplacian in quickstart

**Pull request merged**